compiler-core/src/language_server/tests/definition.rs

```diff
@@ -0,0 +1,367 @@
+use lsp_types::{
+    GotoDefinitionParams, Location, Position, Range, TextDocumentIdentifier,
+    TextDocumentPositionParams, Url,
+};
+
+use super::*;
+
+fn positioned_with_io(
+    src: &str,
+    position: Position,
+    io: &LanguageServerTestIO,
+) -> Option<Location> {
+    let mut engine = setup_engine(io);
+
+    let _ = io.src_module("app", src);
+    for package in &io.manifest.packages {
+        add_package_from_manifest(&mut engine, package.clone());
+    }
+    let response = engine.compile_please();
+    assert!(response.result.is_ok());
+
+    let path = Utf8PathBuf::from(if cfg!(target_family = "windows") {
+        r"\\?\C:\src\app.gleam"
+    } else {
+        "/src/app.gleam"
+    });
+
+    let url = Url::from_file_path(path).unwrap();
+
+    let params = GotoDefinitionParams {
+        text_document_position_params: TextDocumentPositionParams::new(
+            TextDocumentIdentifier::new(url),
+            position,
+        ),
+        work_done_progress_params: Default::default(),
+        partial_result_params: Default::default(),
+    };
+    let response = engine.goto_definition(params);
+
+    response.result.unwrap()
+}
```

Comment:
This looks familiar- is it copied from another test module? If so could we make that one public and use it across both?

Reply:
Good call, yeah this is effectively copied from hover (I thought I'd have to change it more at first). Lemme separate the engine create into a helper func

Reply:
LMK if this is roughly what you had in mind